docs/root/api-v1/http_filters/ip_tagging_filter.rst

```diff
@@ -0,0 +1,63 @@
+.. _config_http_filters_ip_tagging_v1:
+
+HTTP IP Tagging Filter
+======================
+
+HTTP IP Tagging :ref:`configuration overview <config_http_filters_ip_tagging>`.
+
+.. code-block:: json
+
+  {
+    "name": "ip_tagging",
+    "config": {
+      "request_type": "...",
+      "fill_factor" : "...",
+      "root_branching_factor": "...",
+      "ip_tags": []
+    }
+  }
+
+request_type
+  *(optional, string)* The type of requests the filter should apply to. The supported
+  types are *INTERNAL*, *EXTERNAL* or *BOTH*. A request is considered internal if
+  :ref:`x-envoy-internal<config_http_conn_man_headers_x-envoy-internal>` is set to true. If
```

Comment:
Can we link to an authoritative section of the docs on what is internal/external and avoid a long explanation here? I think this idea of internal/external is going to evolve, so would be good to avoid having the IP tagging docs become stale when it does.

Reply:
@ccaraman do you have a use case where the `request_type` would need to be something other than `BOTH`?

Reply:
@htuch Are you thinking of the [xff section](https://www.envoyproxy.io/docs/envoy/latest/configuration/http_conn_man/headers.html?highlight=internal#x-forwarded-for)? The underlying implementation will check the [x-envoy-internal header](https://www.envoyproxy.io/docs/envoy/latest/configuration/http_conn_man/headers.html?highlight=internal#x-envoy-internal) to determine its 'status'. 
What could make this clearer is explicitly calling out that x-envoy-internal is the header used, but that is based on the xff value.

@brian-pane Yes, Lyft use cases will apply different tags based on internal and external.

Reply:
@ccaraman yeah, that came to mind. Basically, just having a single-source-of-truth on what counts as internal/external, everyone links there.

Reply:
I've updated the text and linked it directly to part 2 in things to know about xff: https://www.envoyproxy.io/docs/envoy/latest/configuration/http_conn_man/headers.html?highlight=internal#x-forwarded-for